functions/Set-DbaMaxDop.ps1

```diff
@@ -83,9 +83,8 @@ function Set-DbaMaxDop {
         Set recommended Max DOP for all databases on server sql2016.
 
     #>
-    [CmdletBinding(SupportsShouldProcess)]
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact = 'Medium')]
     param (
-        [parameter(Mandatory, ValueFromPipeline)]
```

Comment:
This should not be removed.

Reply:
No? The pipeline input all comes in the $InputObject and is handled... the error reported is because it is expecting to receive both InputObject and SqlInstance.

Reply:
We do not assume everyone is piping in an object from a command...Pipe in a string and the finding fails:
![image](https://user-images.githubusercontent.com/11204251/66328970-4000c500-e8f3-11e9-84a6-4d521f69ecce.png)

Reply:
With that in place PS binds the value properly...but I also see some other issues with how that InputObject is being handled (but can fix those in another PR)

![image](https://user-images.githubusercontent.com/11204251/66329216-c4ebde80-e8f3-11e9-901b-4215b31b4f20.png)

Reply:
Ah, I wasn't thinking about piping in just the name! Good catch, thank you. The issue was that originally it was Mandatory, added back ValueFromPipeline and it works now. (Also added a test for that!)